lib/puppet/http/client.rb

```diff
@@ -16,7 +16,26 @@ def connect(uri, &block)
     end
   end
 
+  def get(url, headers: {}, params: {})
+    connect(url) do |http|
+      query = encode_params(params).join('&')
+      path = "#{url.path}?#{query}"
+
+      resp = http.get(path, @default_headers.merge(headers))
+      Puppet.info("HTTP GET #{url} returned #{resp.code} #{resp.reason}")
```

Comment:
how is http proxy or server auth handled?

Reply:
Proxy auth is handled when the factory creates the `Net::HTTP` object.

Server auth is not handled currently. Need to account for `URI#user` and `URI#password` containing user auth info.